### ISSUE TITLE
Loop stops looping

### ISSUE THREAD

**Issue by kristinkand** [LoopKit/Loop#2212]
Loop stops looping because it misses the connection to the CGM. Sometimes the connection is reestablished by itself after a while, sometimes not and the app has to be restarted. That happens randomly but frequently.

**To Reproduce**
It just happens. Sometimes several times a day, sometimes not at all.

**Phone**
 - Hardware: iPhone XR
 - OS Version: iOS 17.6.1

**Loop Version**
 - Version Number: 3.5.0 dev
 - Repo: LoopWorkspace, kristinkand

**CGM**
 - Device: Dexcom G7
 - Manager app: Dexcom App

**Pump**
 - Manufacturer: Omnipod
 - Model: Dash[](url)

**Comment by marionbarker** [comment]
The Loop Reports contain 84 hours of data. Next time it happens, please note the time and post your report here. (Loop, Settings, Issue Report)

**Comment by kristinkand** [comment]
I’m sorry, it wasn’t my intention to overwhelm with the loop report. I actually had trouble attaching it since I got an error about the file format.
Anyway I think it relates to the issue on the zulipchat where I’d tried to post it first but since the issue keeps coming back, I came here. 
https://loop.zulipchat.com/#narrow/stream/144111-general/topic/LOOP.20CANNOT.20CONNECT.20WITH.20G7

**Comment by blameitonben** [comment]
Just to confirm, your Dexcom APP will have data (fresh BG’s) but loop is not importing/catching them as they come in every 5 minutes?

I can say we attempted to loop with G7 several months ago, and found it totally impossible. The loop phone could be less than 5 feet away line of site of the sensor and would be in signal loss. We had to shift back to G6

In this case this was not a loop issue but the G7 loosing connection constantly. It sounds like your issue is different from signal loss but wanted to confirm that you G7 app is normally receiving good data from the sensor, while loop is actually missing the readings.

Ben
________________________________

**Comment by kristinkand** [comment]
@blameitonben Yes, that is correct. The Dexcom app shows fresh data but Loop shows “signal loss”. We have been experiencing this since the summer and I'd say since we upgraded to 3.5.0 (we didn’t have this issue when we started looping this spring).

I’ve turned off focus, low power mode, no direct to watch and the kid wasn’t playing a game. It just happens randomly.

**Comment by marionbarker** [comment]
Closing this issue in favor of #2265

The G7 problem has been reported many times and is under investigation. Only keeping one Issue open for tracking purposes.
